### PR TITLE
Adds checks for checking that a directory or a file exists

### PR DESCRIPTION
**Pull request by mik-dass**
Signed-off-by: mik-dass mrinald7@gmail.com
Issue : #52

### CODE REVIEW

File: cmd/create.go [hunk 1]
```diff
@@ -104,6 +104,12 @@ A full list of component types that can be deployed is available using: 'odo com
 			// we want to use and save absolute path for component
 			dir, err := filepath.Abs(componentLocal)
 			checkError(err, "")
+			fi, err := os.Stat(dir)
```
[the remark this answers is not in the document]
Comment (mik-dass) [reply cmd/create.go]: ok :+1:

File: cmd/update.go [hunk 2]
```diff
@@ -90,12 +90,24 @@ var updateCmd = &cobra.Command{
...
+				fmt.Println("Please provide path to a directory")
+				os.Exit(1)
+			}
 			err = component.Update(client, componentName, "local", dir)
 			checkError(err, "")
 			fmt.Printf("The component %s was updated successfully\n", componentName)
 		} else if len(componentBinary) != 0 {
 			path, err := filepath.Abs(componentBinary)
 			checkError(err, "")
+			fileInfo, err := os.Stat(path)
```
Comment (kadel) [inline cmd/update.go]: It should be possible to create binary component even if binary file doesn't exist #436

Comment (mik-dass) [reply cmd/update.go]: What about create.go then?

Comment (kadel) [reply cmd/update.go]: for create.go the same. I actually wanted to post my comment for create.go.

for the local component, we can verify that the directory exists.  But for binary component, it is ok, if the binary file doesn't exist at the time when `odo create` or `odo update` is run. that check should be done only for `odo push`

Comment (mik-dass) [reply cmd/update.go]: OK :+1:

Comment (mik-dass) [reply cmd/update.go]: @kadel What about the isDir thing? Should that be removed too?

**Comment by kadel** [comment]
ping @cdrage

File: cmd/update.go [hunk 3]
```diff
@@ -90,6 +90,12 @@ var updateCmd = &cobra.Command{
 			// we want to use and save absolute path for component
 			dir, err := filepath.Abs(componentLocal)
 			checkError(err, "")
+			fileInfo, err := os.Stat(dir)
+			checkError(err, "")
+			if !fileInfo.IsDir() {
+				fmt.Println("Please provide path to a directory")
```
Comment (cdrage) [inline cmd/update.go]: It should be "Please provide a path to the directory"

Comment (mik-dass) [reply cmd/update.go]: @cdrage Done

**Comment by cdrage** [comment]
LGTM

**Pull request merged**